### DISCUSSION TITLE
Backup laravel pulse tables

### DISCUSSION THREAD

**Discussion by PankovAlxndr** [spatie/laravel-backup#1772] [Q&A]
- "mysqsl": "8",
- "php": "^8.2.0",
- "laravel/framework": "^10.43.0"
- "laravel/pulse": "^1.0@beta",
- "spatie/laravel-backup": "^8.6.0"

I use the laravel-backup for backing up the database, here's the command executed via cron 
`backup:run --only-db --only-to-disk=s3-backup.`

Everything is fine, backups exist, but when I restore a backup 
I get this error: 
`ERROR 3105 (HY000) at line 1116: The value specified for generated column 'key_hash' in table 'pulse_aggregates' is not allowed.`

Interestingly, when I do the backup manually, like this:
`mysqldump --no-tablespaces --opt --user=.... --password=... ... | gzip > /var/www/...../prod.sql.gz,` 
there are no problems.

**Comment by PankovAlxndr** [comment]
?

**Comment by cbaconnier** [comment]
>  mysqldump --no-tablespaces --opt --user=.... --password=... ... | gzip > /var/www/...../prod.sql.gz,

Produce the same error for me.

I believe there isn't a way with `mysqldump` to ignore theses values, neither when importing them with `mysql`. So I don't think this could be fixed in spatie's laravel-backup.
But best would be to use a separate database to store Pulse's data.

If you can't use a separate database and have to restore the dump, easiest way is to remove them from the dump:

```bash
sed '/^INSERT INTO `\(pulse_aggregates\|pulse_values\|pulse_entries\)`/,/;$/d' source.sql > destination.sql
```

_⚠️ Preferably, run the `sed` command before importing and not when exporting in case it would match an unexpected `;`along the way and breaks the rest of the backuped data._
